pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -0,0 +1,789 @@
+			TargetKeyId: &entry.Arn,
+		})
+		if err != nil {
+			p.log.Error("Failed to refresh alias",
+				aliasNameTag, entry.AliasName, keyArnTag, entry.Arn, reasonTag, err)
+		}
+	}
+}
+
+func (p *Plugin) disposeAliasesTask(ctx context.Context) {
+	ticker := p.hooks.clk.Ticker(disposeAliasesFrequency)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping dispose aliases task", reasonTag, ctx.Err())
+			return
+		case <-ticker.C:
+			p.disposeAliases(ctx)
+			p.notifyTaskDone()
+		}
+	}
+}
+
+func (p *Plugin) disposeAliases(ctx context.Context) {
+	p.log.Debug("Looking for aliases in trust domain to dispose")
+	paginator := kms.NewListAliasesPaginator(p.kmsClient, &kms.ListAliasesInput{Limit: aws.Int32(100)})
+
+	for {
+		aliasesResp, err := paginator.NextPage(ctx)
+		switch {
+		case err != nil:
+			p.log.Error("Failed to fetch aliases to dispose", reasonTag, err)
+			return
+		case aliasesResp == nil:
+			p.log.Error("Failed to fetch aliases to dispose: nil response")
+			return
+		}
+
+		for _, alias := range aliasesResp.Aliases {
+			switch {
+			case alias.AliasName == nil || alias.LastUpdatedDate == nil || alias.AliasArn == nil:
+				continue
+				// if alias does not belong to trust domain skip
+			case !strings.HasPrefix(*alias.AliasName, p.aliasPrefixForTrustDomain()):
+				continue
+			// if alias belongs to current server skip
+			case strings.HasPrefix(*alias.AliasName, p.aliasPrefixForServer()):
+				continue
+			}
+
+			now := p.hooks.clk.Now()
+			diff := now.Sub(*alias.LastUpdatedDate)
+			if diff < aliasThreshold {
+				continue
+			}
+			log := p.log.With(aliasNameTag, alias.AliasName)
+			log.Debug("Found alias in trust domain beyond threshold")
+
+			describeResp, err := p.kmsClient.DescribeKey(ctx, &kms.DescribeKeyInput{KeyId: alias.AliasArn})
+			switch {
+			case err != nil:
+				log.Error("Failed to describe key to dispose", reasonTag, err)
+				continue
+			case describeResp == nil || describeResp.KeyMetadata == nil || describeResp.KeyMetadata.Arn == nil:
+				log.Error("Malformed describe key response while trying to dispose")
+				continue
+			case !describeResp.KeyMetadata.Enabled:
+				continue
+			}
+			log = log.With(keyArnTag, *describeResp.KeyMetadata.Arn)
+
+			_, err = p.kmsClient.DeleteAlias(ctx, &kms.DeleteAliasInput{AliasName: alias.AliasName})
+			if err != nil {
+				p.log.Error("Failed to delete alias to dispose", reasonTag, err)
+				continue
+			}
+
+			select {
+			case p.scheduleDelete <- *describeResp.KeyMetadata.Arn:
+				log.Debug("Key enqueued for deletion")
+			default:
+				log.Error("Failed to enqueue key for deletion")
+			}
+		}
+
+		if !paginator.HasMorePages() {
+			break
+		}
+	}
+}
+
+func (p *Plugin) disposeKeysTask(ctx context.Context) {
+	ticker := p.hooks.clk.Ticker(disposeKeysFrequency)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			p.log.Debug("Stopping dispose keys task", reasonTag, ctx.Err())
```

Comment:
These kinds of log messages feel a little over the top. I say that because they're useful to developers, but not very useful to an operator. If the context is cancelled, there will be lots of other stuff in the log, and what the dispose keys task is doing doesn't feel very important to know (even if it's debug log)